src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -146,11 +146,7 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
                 throw new ArgumentNullException("selector");
             }
 
-            //It is certain the result is already ready here and can be collected straight away. Having a truly asynchronous result collection
-            //IAsyncEnumerable<TResult> or equivalent method ought to be used. Taking the result here without async-await saves for generating
-            //the async state machine.
-            var ret = ExecuteAsync(query, parameterProvider, ExecuteReaderAsync, selector, cancellationToken, commandBehavior).GetAwaiter().GetResult().Item1;
```

Comment:
Well, we still have other test failures that we must sort our, but wanted your confirmation first. Maybe the new failures are because it is now inserting so much faster given that it's non-blocking (previously if it was blocking the Orleans threads, it would have been limiting concurrency).